src/net_processing.cpp

```diff
@@ -3944,6 +4160,163 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         return;
     }
 
+    std::chrono::microseconds current_time = GetTime<std::chrono::microseconds>();
```

Comment:
Please place this inside the `REQRECON` message handling code block (we probably want to split each message handler into its own function in the future, so we should minimize the use of code outside those code blocks.